utilsplugin2/src/org/openstreetmap/josm/plugins/utilsplugin2/actions/CopyWithRelationAction.java

```diff
@@ -0,0 +1,102 @@
+// License: GPL. For details, see LICENSE file.
+// Author: David Earl
+package org.openstreetmap.josm.plugins.utilsplugin2.actions;
+
+import static org.openstreetmap.josm.gui.help.HelpUtil.ht;
+import static org.openstreetmap.josm.tools.I18n.tr;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.KeyEvent;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+
+import javax.swing.JOptionPane;
+
+import org.openstreetmap.josm.Main;
+import org.openstreetmap.josm.actions.JosmAction;
+import org.openstreetmap.josm.data.osm.DataSet;
+import org.openstreetmap.josm.data.osm.OsmPrimitive;
+import org.openstreetmap.josm.data.osm.Relation;
+import org.openstreetmap.josm.gui.MainApplication;
+import org.openstreetmap.josm.gui.datatransfer.ClipboardUtils;
+import org.openstreetmap.josm.gui.datatransfer.PrimitiveTransferable;
+import org.openstreetmap.josm.gui.datatransfer.data.PrimitiveTransferData;
+import org.openstreetmap.josm.gui.layer.OsmDataLayer;
+import org.openstreetmap.josm.tools.Shortcut;
+
+/**
+ * Copy OSM primitives and relation to clipboard in order to paste them, or their tags, somewhere else.
+ * @since 404
+ */
+public class CopyWithRelationAction extends JosmAction {
+    /**
+     * Constructs a new {@code CopyAction}.
+     */
+    public CopyWithRelationAction() {
+        super(tr("Copy Relation"), "copy",
+                tr("Copy selected objects and relations to paste buffer."),
+                Shortcut.registerShortcut("system:copy:relation", tr("Edit: {0}", tr("Copy Relation")), KeyEvent.VK_C, Shortcut.ALT_SHIFT), true);
+        putValue("help", ht("/Action/Copy"));
+        // CUA shortcut for copy (https://en.wikipedia.org/wiki/IBM_Common_User_Access#Description)
+        MainApplication.registerActionShortcut(this,
+                Shortcut.registerShortcut("system:copy:relation:cua", tr("Edit: {0}", tr("Copy Relation")), KeyEvent.VK_INSERT, Shortcut.CTRL_SHIFT));
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        DataSet set = getLayerManager().getEditDataSet();
+        Collection<Relation> relations = set.getRelations();
```

Comment:
wouldn't it be more performant to get use `getParentRelations(selection)` instead of the whole dataset?

Reply:
Pretty sure I tried that and it didn't work.